src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -633,12 +649,157 @@ public Snapshot makeSnapshotFromSnapshotRequest(SnapshotRequestModel snapshotReq
         .tags(TagUtils.sanitizeTags(snapshotRequestModel.getTags()));
   }
 
-  private void createSnapshotTablesFromDatasetAsset(
-      String datasetName, String assetName, SnapshotSource snapshotSource, Snapshot snapshot) {
-    Dataset queryDataset = datasetService.retrieveByName(datasetName);
-    AssetSpecification queryAssetSpecification = getAssetByNameFromDataset(queryDataset, assetName);
-    snapshotSource.assetSpecification(queryAssetSpecification);
-    conjureSnapshotTablesFromAsset(snapshot, snapshotSource);
+  private SnapshotAccessRequestResponse getById(UUID accessRequestId) {
+    return snapshotRequestDao.getById(accessRequestId);
+  }
+
+  @VisibleForTesting
+  AssetSpecification buildAssetFromSnapshotAccessRequest(
+      Dataset dataset, SnapshotAccessRequestResponse snapshotRequestModel) {
+    // build asset model from snapshot request
+    AssetModel assetModel = new AssetModel();
+    assetModel.name("snapshot-by-request-asset");
+    assetModel.rootTable("person");
+    assetModel.rootColumn("person_id");
```

Comment:
I think it's clearer to write this using fluent-style syntax
```suggestion
    AssetModel assetModel = new AssetModel()
        .name("snapshot-by-request-asset")
        .rootTable("person")
        .rootColumn("person_id");
```